pkg/cluster/internal/delete/delete.go

```diff
@@ -35,15 +35,16 @@ func Cluster(logger log.Logger, p providers.Provider, name, explicitKubeconfigPa
 
 	kerr := kubeconfig.Remove(name, explicitKubeconfigPath)
 	if kerr != nil {
-		logger.Errorf("failed to update kubeconfig: %v", kerr)
+		return kerr
+	} else {
+		logger.V(0).Infof("Removed context %q from kubeconfig", kubeconfig.ContextForCluster(name))
 	}
 
 	err = p.DeleteNodes(n)
 	if err != nil {
 		return err
-	}
-	if kerr != nil {
-		return err
+	} else {
+		logger.V(0).Infof("Deleted nodes: %q", n)
```

Comment:
We shouldn't log this when len(nodes) = 0

Reply:
How can I test a cluster with no nodes?

Reply:
@BenTheElder I don't see how this would ever trigger.

Reply:
Ping @BenTheElder

Reply:
this is a public API, so anyone can call it against this for a cluster name even if there are no nodes

Reply:
specifically by calling https://github.com/kubernetes-sigs/kind/blob/d53225750af998c6218b5c8c730b8b3ca0e2d472/pkg/cluster/provider.go#L186-L188

Reply:
we can check len(nodes) and avoid the deleteNodes call and the logging when there are no nodes